endpoints/s3/src/putobject.cpp

```diff
@@ -198,19 +309,20 @@ class incremental_async_read : public std::enable_shared_from_this<incremental_a
 					self->session_ptr_->send(std::move(self->resp_)); // Schedules an async write op.
 					return;
 				}
-				logging::debug(
+				logging::trace(
 					"{}: multipart upload: Wrote [{}] bytes to iRODS file [{}].",
 					__func__,
 					byte_count,
 					self->irods_path_);
 			}
 
 			if (self->parser_->is_done()) {
-				if (self->upload_part_flag_) {
+				if (self->part_file_.is_open()) {
```

Comment:
Just making sure... The only time `part_file_.is_open()` will be `true` is if `upload_part_flag_` is `true` as well, right?

Reply:
Yes that is true.  This code also handles non-multipart uploads.

Also, even with `upload_part_flag_ = false`, the part file won't be open if we wrote that part directly to iRODS.